ui/src/services/NodeVolumesUtils.js

```diff
@@ -168,13 +168,26 @@ export const getVolumeListData = createSelector(
     volumeLatencyCurrent,
     volumeCapacityCurrentList,
   ) => {
-    let nodeVolumes = volumes;
-    // filter the volumes by the node name from URL
-    if (nodeName) {
-      nodeVolumes = volumes?.filter(
-        (volume) => volume.spec.nodeName === nodeName,
-      );
+    let nodeVolumes = null;
+
+    if (nodeFilter) {
+      // when there is node filter in URL, check if the node name is valid / exist
```

Comment:
If this case is really important to consider, shouldn't we treat it more explicitly for the user? Otherwise, from a volume listing standpoint, it is the same to filter by a node which has no volume, or a non-existing node (since we just return an empty list).

Reply:
Absolutely right! 
Since we return an empty array for both cases, a node which has no volume and a non-existing node, I shouldn't implement such a complex logic here!